From 928c89f75da9e44559a67ae83041c06dba5a4cb1 Mon Sep 17 00:00:00 2001
From: Emil Jahshan <emil.jahshan@intel.com>
Date: Thu, 7 Oct 2021 11:44:08 +0300
Subject: [PATCH] d4xx: fixing RGB format report

Signed-off-by: Emil Jahshan <emil.jahshan@intel.com>
---
 drivers/media/i2c/d4xx.c | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/drivers/media/i2c/d4xx.c b/drivers/media/i2c/d4xx.c
index 5d3653468..aca8b3527 100644
--- a/drivers/media/i2c/d4xx.c
+++ b/drivers/media/i2c/d4xx.c
@@ -2144,6 +2144,12 @@ static int ds5_mux_get_fmt(struct v4l2_subdev *sd, struct v4l2_subdev_pad_config
 
 	dev_info(sd->dev, "%s(): %u %p\n", __func__, pad, state->mux.last_set);
 
+	if (state->is_rgb) {
+		tmp.pad = DS5_MUX_PAD_RGB;
+		pad = DS5_MUX_PAD_RGB;
+	} else
+		tmp.pad = 0;
+
 	switch (pad) {
 	case DS5_MUX_PAD_MOTION_T:
 		remote_sd = &state->motion_t.sensor.sd;
@@ -2154,11 +2160,13 @@ static int ds5_mux_get_fmt(struct v4l2_subdev *sd, struct v4l2_subdev_pad_config
 	case DS5_MUX_PAD_EXTERNAL:
 		remote_sd = &state->mux.last_set->sd;
 		break;
+	case DS5_MUX_PAD_RGB:
+		remote_sd = &state->rgb.sensor.sd;
+		break;
 	default:
 		return -EINVAL;
 	}
 
-	tmp.pad = 0;
 
 	/* Locks internally */
 	ret = ds5_sensor_get_fmt(remote_sd, cfg, &tmp);
-- 
2.17.1

